django/db/backends/mysql/base.py

```diff
@@ -341,6 +342,7 @@ def data_type_check_constraints(self):
             return {
                 'PositiveIntegerField': '`%(column)s` >= 0',
                 'PositiveSmallIntegerField': '`%(column)s` >= 0',
+                'JSONField': 'JSON_VALID(%(column)s)',
```

Comment:
Small improvement - I think this is only needed on MariaDB - [MySQL docs](https://dev.mysql.com/doc/refman/8.0/en/json.html) say:

> Automatic validation of JSON documents stored in JSON columns. Invalid documents produce an error.

This would avoid double-validation of JSON columns on MySQL

Reply:
Good call. It's also automatically used on MariaDB >= 10.4.3, so I changed it accordingly. I'm not sure if the way I wrote it is acceptable, though. I think it's cleaner than assigning the dictionary to a variable first, but it might be a bit too pythonic.

Reply:
Oh, I just remembered. It's wrapped in a

```python
if self.features.supports_column_check_constraints:	
```

so even without the fix, (for now) it's actually only applied on MariaDB :grin:

At least now it's not applied on >= 10.4.3.

Reply:
MySQL 8 [supports column check constraints](https://dev.mysql.com/doc/refman/8.0/en/create-table-check-constraints.html), Django just needs to catch up :)

Reply:
Whoops, I remember reading that! I guess someone has to patch [this line](https://github.com/django/django/blob/master/django/db/backends/mysql/features.py#L92), but I don't know if that's enough.

Reply:
I added a ticket [30750](https://code.djangoproject.com/ticket/30750) and started to work on a patch, but still a lot of schema tests are failing.

Reply:
We added support for check constraints on MySQL 8.0.16+ e2c6a0858d7d9ad85eda353076a5b46608b704a9

Reply:
Cool!